runtime/interpreter/value.go

```diff
@@ -16379,7 +16379,8 @@ type CompositeField struct {
 	Name  string
 }
 
-const attachmentNamePrefix = "$"
+const unrepresentableNamePrefix = "$"
+const resourceDefaultDestroyEventPrefix = "ResourceDestroyed" + unrepresentableNamePrefix
```

Comment:
The default destroy event is encoded like a function on resources, since it behaves differently than other type definitions that can appear inside e.g. interfaces. This is used to generate a guaranteed fresh name for that function to be used when looking up the event constructor in the resource's function map.